### PR TITLE
Some misc README cleanup

### PR DESCRIPTION
**Pull request by lrettig**
Saving work in progress. Going to try to merge some in flight PRs and rebase against those.

### CODE REVIEW

**Comment by axic** [comment]
#96 was merged, please rebase.

File: README.md [hunk 1]
```diff
@@ -36,37 +37,40 @@ The team also hosts a (roughly) fortnightly public "Ewasm community hangout" cal
...
+One of the [design goals](https://github.com/ewasm/design/blob/master/rationale.md) of the Ewasm project is to support smart contract development in a wide range of languages, using a range of tooling, including existing, mature toolchains such as LLVM (C/C++, Kotlin, Rust, and [many others](https://en.wikipedia.org/wiki/LLVM)) and JavaScript/TypeScript. In theory, any language that can be compiled to Wasm can be used to write a smart contract (as long as it implements the [contract interface](https://github.com/ewasm/design/blob/master/contract_interface.md) and [Ethereum interface](https://github.com/ewasm/design/blob/master/eth_interface.md)). See [awesome-wasm-langs](https://github.com/appcypher/awesome-wasm-langs) for a list of such languages.
 
 At present, we've developed support for the following languages and toolchains:
 
-- LLVM: C, C++, and Rust: documentation pending
 - [AssemblyScript](https://github.com/AssemblyScript/assemblyscript), a subset of TypeScript, which uses the JavaScript toolchain: see the [etherts org](https://github.com/etherts/docs) for more information on writing contracts in AssemblyScript.
+- LLVM: C, C++, and Rust: documentation pending
```
Comment (axic) [inline README.md]: Don't agree with the swap entirely, Rust is actually working :)

**Comment by lrettig** [comment]
Fair enough but it has no documentation that I'm aware of.

File: README.md (hunk 1 above)
Comment (lrettig) [reply README.md]: Fair enough but it has no documentation that I'm aware of.

Comment (axic) [reply README.md]: Self documenting code! Honestly, @jakelang added that to ewasm-rust-api. Rustdoc can be generated.

Comment (lrettig) [reply README.md]: @jakelang if you can share a link I'll add it to the doc (and put Rust on top :P)

**Comment by lrettig** [comment]
This is ready to go but will likely conflict with #115

**Comment by axic** [comment]
Rebased - because it was hard to follow what is going on with merge commits.

**Comment by lrettig** [comment]
From my perspective this is ready to merge, @axic ?

**Comment by lrettig** [comment]
Is this ready to go? Ping

**Review by chfast — DISMISSED** [review]
Squash?

**Comment by axic** [comment]
I'm not sure why wagon was moved out of the main doc.

**Comment by lrettig** [comment]
> I'm not sure why wagon was moved out of the main doc.

I moved it into a side doc because it's not ready for production yet. I did the same thing for aleth. Would prefer to keep the master doc as focused and simple as possible. @axic is there a reason you want it in the main doc?

**Comment by axic** [comment]
I've pulled out two major changes from here and merged them separately.

> is there a reason you want it in the main doc?

Wagon is ready and passing the tests (as of yesterday). It is planned to be the main client for 2nd milestone. Don't wanna move it back and forth.

**Comment by lrettig** [comment]
Then can we remove the part that says "this spec may be wrong and will
change"?

**Comment by axic** [comment]
@gballet is wagon working now properly? Can we remove the warning?

File: README.md [hunk 2]
```diff
@@ -158,55 +158,37 @@ Aleth supports EVMC out of the box, but geth not yet.
...
+	Here's the recommended configuration for running a node locally:
 
 	```sh
 	> ./build/bin/geth \
 	--vm.ewasm="/path/to/libhera.so,metering=true,fallback=true" \
 	--datadir /tmp/ewasm-node/4201/ \
-	--etherbase 031159dF845ADe415202e6DA299223cb640B9DB0 \
 	--rpc --rpcapi "web3,net,eth,debug" \
-	--rpcvhosts="*" --rpcaddr "0.0.0.0" \
+	--rpcvhosts=* --rpcaddr "0.0.0.0" \
```
Comment (axic) [inline README.md]: I think this needs to stay in quotes otherwise the shell expands it.

Comment (lrettig) [reply README.md]: Fixed

**Comment by axic** [comment]
@gballet ping https://github.com/ewasm/testnet/pull/102#issuecomment-441668640

**Comment by gballet** [comment]
@axic wagon is working with some issue that I found, need to create the PR for the fix and have them integrate them.

File: README.md (hunk 1 above)
Comment (axic) [reply README.md]: Checked the link to assemblyscript, but I couldn't find extensive documentation explaining how to compile/use.  How is it better documented? :)

This is kind of moot after #129 is merged.

File: README.md [hunk 3]
```diff
@@ -168,102 +159,76 @@ Aleth supports EVMC out of the box, but geth not yet.
...
-	> ./geth \
-	--datadir /tmp/ewasm-node/4201/ \
-	--etherbase 031159dF845ADe415202e6DA299223cb640B9DB0 \
-	--rpc --rpcapi "web3,net,eth,debug" \
-	--rpcvhosts="*" --rpcaddr "0.0.0.0" \
-	--rpccorsdomain "*" \
-	--mine --miner.threads 1 \
-	--nodiscover \
+	--etherbase a8c3eeb2915373139bcfc287d4ae9e660d734881 \
 	--networkid 66 \
```
Comment (axic) [inline README.md:227]: Why is it better to split these two instructions? It seems weird that this block only contains extra options.

Comment (lrettig) [reply README.md:227]: Are you referring to the `geth` configuration specified at [line 174](https://github.com/ewasm/testnet/pull/102/files#diff-04c6e90faac2675aa89e2176d2eec7d8R174)? Because there's a difference between running a node locally and running a node that syncs the testnet -- these additional commands are intended for testnet syncing. (Though I think `--etherbase` could be removed)

**Comment by lrettig** [comment]
Did you remove the assemblyscript docs via force push? 😑

**Comment by axic** [comment]
> Did you remove the assemblyscript docs via force push? 😑

Didn't remove any assemblyscript docs.

File: README.md (hunk 3 above)
Comment (axic) [reply README.md:227]: What is the use of "running a node locally" without connecting to the testnet?

FYI running the node locally can be done in "test" mode which avoids 90% of the commandline options.

Comment (lrettig) [reply README.md:227]: > What is the use of "running a node locally" without connecting to the testnet?

Local development.

> FYI running the node locally can be done in "test" mode which avoids 90% of the commandline options.

Okay. Presumably we still need the `--vm.ewasm` flag but not the RPC-related ones?

**Comment by lrettig** [comment]
Can we just merge this, then update the wagon stuff later as necessary?

**Comment by axic** [comment]
> Can we just merge this, then update the wagon stuff later as necessary?

There is little point applying all these wagon changes from this PR if we need to roll it back 100%. Wagon just works.

**Comment by lrettig** [comment]
It would be great to get this merged, and iterate. There's a bunch of confusing/misleading stuff in the master doc right now that this fixes.

**Comment by axic** [comment]
> There's a bunch of confusing/misleading stuff in the master doc right now that this fixes.

Can you please list what is this fixing? The C++/Rust swap is not relevant anymore, Wagon is not experimental anymore.

**Comment by lrettig** [comment]
This PR is not about C++/Rust or Wagon. It's about misc doc cleanup. The bulk of these changes are still relevant. Is there some specific concern here? If not, can we merge and iterate?
